build/tools/utils.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+export PS4='\e[36m+ ${FUNCNAME:-main}@${BASH_SOURCE}:${LINENO} \e[0m'
+
+function check_deps() {
+  local dependencies=("$@")
+  echo "Checking dependencies..."
+
+  for dep in ${dependencies[@]}; do
+    if ! command -v $dep &>/dev/null; then
+      echo "Please install $dep"
+      exit 1
+    fi
+  done
+
+  echo "All dependencies are installed"
+}
+
+function strip_prefix() {
+  local str="$1"
+  local prefix="$2"
+
+  echo ${str#"$prefix"}
+}
+
+function version_without_v() {
+  strip_prefix "$1" v
+}
+
+function version_with_v() {
+  local without_v=$(version_without_v "$1")
+
+  echo "v$without_v"
+}
+
+# get_noobaa_version returns the NooBaa version by reading the `cmd/version/main.go`
+#
+# Note: If the second argument is set to 1, it will append "v" to the version while no 
+# argument will return the version without "v"
+function get_noobaa_version() {
+  local version=$(go run cmd/version/main.go)
+  if [[ $# -eq 1 ]]; then
+    if [[ "$1" == "1" ]]; then
+      version_with_v "$version"
+    else
+      version_without_v "$version"
+      echo "[Warn]: Unknown argument passed to get_noobaa_version."
+    fi
+  else
+    version_without_v "$version"
+  fi
+}
```

Comment:
It is an internal function so we can be not strict about it, but in general, I don't like the `if [[ $# -eq 1 ]]`
basically, it assumes that there will be a max of one argument.
but in the case of `get_noobaa_version 1 2` the intention is `version_with_v` as we got `1` in the first argument, but it will never get into the if.
If we want to be strict about it then we can add something like:
```bash
if [[ $# -gt 1 ]] 
then
    echo "[Error]: expecting one argument at the most" 
fi
```

Reply:
Done